packages/utils/src/useFilterConfig/useFilterConfig.js

```diff
@@ -0,0 +1,37 @@
+import isPlainObject from 'lodash/isPlainObject';
+import useSelectionManager from '../useSelectionManager';
+import { toFilterConfig, getFilterConfigItem } from './filterConfigHelpers';
+import { toFilterChips, toDeselectValue } from './filterChipHelpers';
+
+const useFilterConfig = (options = {}) => {
+  const { filters, onDeleteFilter, resetOnClear } = options;
+  const enableFilters = !!filters;
+  const { filterConfig = [], activeFilters: initialActiveFilters } = filters || {};
+  const { selection: activeFilters, set, deselect, reset, clear } = useSelectionManager(initialActiveFilters, { withGroups: true });
+
+  const onFilterUpdate = (filter, value) => {
+    const configItem = getFilterConfigItem(filterConfig, filter);
+    const passValue = configItem.type === 'text' && value.length === 0 ? undefined : [value];
+    set(Array.isArray(value) || isPlainObject(value) ? value : passValue, filter);
+  };
+
+  const onFilterDelete = async (_event, chips, clearAll = false) => {
+    clearAll ? (resetOnClear ? reset() : clear()) : deselect(...toDeselectValue(filterConfig, chips[0]));
+    onDeleteFilter && onDeleteFilter(chips, clearAll);
```

Comment:
Safe accessor can be used here
```suggestion
    onDeleteFilter?.(chips, clearAll);
```